core-contracts/Dex/src/main/java/network/balanced/score/core/dex/AbstractDex.java

```diff
@@ -600,6 +600,7 @@ void exchange(Address fromToken, Address toToken, Address sender,
         BigInteger fees = lpFees.add(balnFees);
 
         Address poolBaseToken = poolBase.get(id);
+        Address poolQuoteToken = poolQuote.get(id);
         boolean isSell = fromToken.equals(poolBaseToken);
```

Comment:
we can avoid the additional getter as well if we want to optimize slightly
```suggestion
        boolean isSell = fromToken.equals(poolBaseToken);
        Address poolQuoteToken = isSell ? toToken : fromToken
```